src/ring_theory/algebra.lean

```diff
@@ -286,13 +287,21 @@ instance linear_map.semimodule' (R : Type u) [comm_semiring R]
 end semiring
 
 section ring
-variables [comm_ring R] [comm_ring S] [ring A] [algebra R A]
+variables [comm_ring R]
 
-lemma mul_sub_algebra_map_commutes (x : A) (r : R) :
+/-- A semiring that is an algebra over a commutative ring carries a natural ring structure. -/
+def as_ring [semiring A] [algebra R A] : ring A :=
```

Comment:
I think your original name `ring_of_comm_ring_algebra` is far better than my `algebra.as_ring`.

Reply:
It's not actually my name, I think I preferred yours!

Note the current name is `algebra.as_ring` since this is within the algebra namespace. Another option would be to call this `semi_ring.asring`, or use `to_` instead of `as` to match `semiring.to_semimodule`

Reply:
Wait, this actually belongs in module right, as
```
example (R : Type) (M : Type) [ring R] [add_comm_monoid M] [semimodule R M] : add_comm_group M
```